src/main/kotlin/com/jetbrains/snakecharm/lang/parser/SmkStatementParsing.kt

```diff
@@ -289,13 +318,298 @@ class SmkStatementParsing(
+
+        when (myBuilder.tokenType) {
+            PyTokenTypes.FROM_KEYWORD -> myBuilder.apply {
+                error(SnakemakeBundle.message("PARSE.use.names.expected"))
+                asKeywordExists = fromSignatureParsing(names)
+            }
+            PyTokenTypes.AS_KEYWORD -> myBuilder.apply {
+                asKeywordExists = true
+                error(SnakemakeBundle.message("PARSE.use.names.expected"))
+                asSignatureParsing(names)
+            }
+            PyTokenTypes.IDENTIFIER -> {
+                val marker = myBuilder.mark()
+                parseIdentifierFromIdentifiersList(names)
+                marker.done(SmkElementTypes.USE_IMPORTED_RULES_NAMES)
+                asKeywordExists = endOfImportedRulesDeclaration(false, names)
+            }
+            PyTokenTypes.MULT -> {
+                val marker = myBuilder.mark()
+                names.add(myBuilder.tokenText!!)
+                nextToken()
+                if (myBuilder.tokenType == PyTokenTypes.COMMA) {
+                    myBuilder.error(SnakemakeBundle.message("PARSE.use.wildcard.in.names.list"))
+                    nextToken()
+                    parseIdentifierFromIdentifiersList(names)
+                }
+                marker.done(SmkElementTypes.USE_IMPORTED_RULES_NAMES)
+                asKeywordExists = endOfImportedRulesDeclaration(true, names)
+            }
+            else -> myBuilder.apply {
+                error(SnakemakeBundle.message("PARSE.use.names.expected"))
+            }
+        }
+
+        return asKeywordExists
+    }
+
+    /**
+     * Uses when we need to parse list of imported rules names.
+     */
+    private fun parseIdentifierFromIdentifiersList(
+        list: MutableList<String>
+    ) {
+        var hasNext = true
+        while (hasNext) {
+            hasNext = when (myBuilder.tokenType) {
+                PyTokenTypes.IDENTIFIER -> {
+                    val referenceMarker = myBuilder.mark() // Register new name
+                    list.add(myBuilder.tokenText ?: return)
+                    Parsing.advanceIdentifierLike(myBuilder)
+                    referenceMarker.done(SmkElementTypes.REFERENCE_EXPRESSION)
+                    registerCommaOrEndOfNames()
+                }
+                PyTokenTypes.MULT -> {
+                    myBuilder.error(SnakemakeBundle.message("PARSE.use.wildcard.in.names.list"))
+                    nextToken()
+                    registerCommaOrEndOfNames()
+                }
+                PyTokenTypes.COMMA -> {
+                    myBuilder.error(SnakemakeBundle.message("PARSE.use.names.expected"))
+                    nextToken()
+                    true
+                }
+                // Actually, Snakemake allows any name for rules and modules,
+                // that doesn't have python token type NAME, which may contains such words as:
+                // 'use', 'as', 'from'
+                // Do we need to add such support?
```

Comment:
I think we don't need such support now, thx for the comment let's leave it for future =)